servers/assessment/competencies/competencyDTO/updateCompetencyRequest.go

```diff
@@ -0,0 +1,10 @@
+package competencyDTO
+
+import "github.com/google/uuid"
+
+// UpdateCompetencyRequest represents the request body for updating a competency.
+type UpdateCompetencyRequest struct {
+	Name              string     `json:"name"`
+	Description       string     `json:"description"`
+	SuperCompetencyID *uuid.UUID `json:"superCompetencyId,omitempty"`
```

Comment:
```suggestion
	SuperCompetencyID *uuid.UUID `json:"superCompetencyID,omitempty"`
```

I'm still not a super big fan of *uuid.UUID but lets discuss this on Thursday.